### ISSUE TITLE
Roxygen comments for spin() in loops

### ISSUE THREAD

**Issue by rfenouil** [yihui/knitr#1539]
Dear author(s),

my apologies if opening an issue is not appropriate for my request. I am facing a situation that has already been raised on stackoverflow, but no satisfying solution has been proposed (in my opinion).

knitr::spin() function could be the perfect option for the generation of reports from complex R scripts (nested loops, etc). However, my attempts to use ROxygen comments in a loop failed with an error:

````
for(i in 1:10){
#' blabla  
}
````

In the resulting HTML:
````
for(i in 1:10){
## Error: <text>:8:0: unexpected end of input
## 6: 
## 7: for(i in 1:10){
##   ^
````
blabla
````
}
## Error: <text>:1:1: '}' inattendu(e)
## 1: }
##     ^
````

On Stack Overflow, I could only find 2 discussions related to this issue:
https://stackoverflow.com/questions/28550747/knitr-spin-with-loop-and-comparison-statements
https://stackoverflow.com/questions/17178831/generate-markdown-comments-within-for-loop

The second thread gives a solution using the following trick:
````
#+ results='asis'
for(i in 1:10){
cat("#' blabla  ")
}
````
It seems acceptable, but it breaks the general (and great) concept of using simple ROxygen comments for the report organzation (it forces to change coding style when using a loop).

Because I don't understand the background mechanisms of spin() function, I wonder whether there is a technical limitation that prevent the interpretation of ROxygen comments in loops, or if this could be integrated in future releases.

Thank you for your feedback on this issue.

All the best.

**Comment by yihui** [comment]
Yes, there is a technical limitation -- you cannot insert text into incomplete R code chunks. In other words, `#'` can only appear between two complete blocks of R code.

The solution `cat()` you mentioned is the only way out here.

This is by design, and will be technically very difficult to change, although I can clearly see its value.

**Comment by rfenouil** [comment]
Dear Yihui,

thank you very much for your answer. I now understand why we cannot use ```#'``` in loops.
Despite the technical challenge I hope that you will consider integrating this feature at some point in the future. That would definitely make ```Spin()``` function the utimate method to turn scripts into reports.

If I understand correctly, the only method that (currently) allows seamless integration with loop is brew from pandoc package ?
I found that the syntax you introduced with ROxygen is way more user-friendly though.

Thank you again for your help,

All the best.

**Comment by yihui** [comment]
Yes, the brew package is the only implementation that I'm aware of.

Actually I did consider this in the very beginning (#455) when I was designing knitr, but gave up since it was too complicated (at least to me at that time; it is probably no longer as difficult to implement for me now).

**Comment by rfenouil** [comment]
Ok, thank you very much for your feedback.

I hope that you will have time (and motivation) to accomplish that at some point. The ROxygen comment method is less disruptive and much more intuitive than pandoc approach for report generation.
I'll promote it around me if that can give you motivation :)

More seriously, I'm sure that it could change the way people write R scripts and create reports.
Thank you for your work btw.

All the best.

**Comment by yihui** [comment]
Apologies for the long silence on this one. As discussed above, this is a fundamental limitation: `spin()` treats `#'` lines as text that separates surrounding code into complete chunks, so such a comment inside a loop body (or any incomplete block) would split the code into syntactically invalid pieces. The `cat()` trick with `#+ results='asis'` remains the way to emit text from within a loop. Closing as by-design.
